sys/net/gnrc/doc.txt

```diff
@@ -0,0 +1,172 @@
+/**
+ * @defgroup    net_gnrc Generic (gnrc) network stack.
+ * @ingroup     net
+ * @brief       RIOT's modular default IP network stack.
+ *
+ * About
+ * =====
+ *
+ * This module is currently the default network stack for RIOT and includes many components
+ * ranging from a simple MAC protocol (@ref net_gnrc_nomac) through a fully-featured IPv6
+ * implementation (@ref net_gnrc_ipv6) with 6LowPAN extensions (@ref net_gnrc_sixlowpan) to
+ * an UDP implementation (@ref net_gnrc_udp) and @ref net_gnrc_rpl.
+ *
+ * A list of all features contained in the @ref net_gnrc is available in the `Modules` section above.
+ *
+ * Integration into RIOT
+ * =====================
+ *
+ * From the application layer the @ref net_gnrc can be accessed through the @ref net_gnrc_conn,
+ * while the interface to the @ref drivers_netdev_netdev2 is defined by the @ref net_gnrc_netdev2.
+ *
+ * Architecture
+ * ============
+ *
+ * ![GNRC Network Stack](riot-gnrc.svg)
+ * Each layer of the network stack runs in its own thread and each lower layer thread has a higher
+ * priority than any upper layer thread. In this regard, the thread of the MAC layer
+ * implementation has the highest priority and threads on the application layer have
+ * the lowest priority. The communication between threads is handled by the kernel's
+ * @ref core_msg functionality and by the @ref net_gnrc_netapi. Most of the times inter-process
+ * communication will take place between threads of neighboring layers for packets that traverse
+ * the network stack up or down.
+ *
+ * Due to the design of @ref net_gnrc "GNRC" and the nature of inter-process communication,
+ * it is crucial for a new module that introduces a new thread to follow a certain programming
+ * construct if it desires to interact with other threads without blocking the system:
+ * Utilizing an `event loop`.
+ *
+ * Hence, a thread for @ref net_gnrc "GNRC" will usually consist of four basic steps.
+ * -# Initialize a message queue
```

Comment:
You can use [numbered lists](https://www.stack.nl/~dimitri/doxygen/manual/markdown.html#mddox_lists). This notation is historical.